### PR TITLE
Issue/2043 update stats banner

### LINKED ISSUE #2043: Stats banner update for Woo 4.0
There are a couple of changes needed in the app now that Woo 4.0 is released and includes WooCommerce Admin plugin.

1. Site does not have v4 wc-analytics namespace available OR Woo version is less than 4.0 - inform user that old stats will eventually go away, upgrade to 4.0 or install WCAdmin.

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #2043 by changing the banner message design when stats v4 is not available. I decided to merge this into the `dark-mode` branch since it required some design changes which might be incompatible when dark mode feature is live.

##### When stats is available
. . .  

##### When stats is no longer available

. . . 

#### To test
- Uninstall the app and install the changes from this PR.
- Login to a site with WooCommerce greater than 4.0 or WooCommerce Admin installed.
- Notice that after login, the Stats availability banner is displayed.
- Uninstall the WC-Admin from your site or downgrade the WooCommerce plugin to a version less than 4.0.
- Open the app again and notice that the Stats Reverted banner is displayed.
- Clicking on `Learn More` button opens the WooCommerce plugin page.
- Clicking on the `Dismiss` button dismisses the banner.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by Garance91540** [comment]
Hi @anitaa1990, 

As we are planning to retire old stats around July, I was thinking of replacing the ["try our improved stats"](https://d.pr/i/IowqFG) banner with ["upgrade to keep seeing your stats"](https://d.pr/i/F168I3)
cc @astralbodies 

Where does the "Learn More" CTA takes you on the "upgrade to keep stats" banner? 

Design reference: https://woomobilep2.wordpress.com/2020/02/19/stats-banner-wc-4-0/#comment-207

**Comment by anitaa1990** [comment]
Hey @Garance91540, thanks for the review! I was following the designs from the Figma file. Adding the design here for some clarity:



It is mentioned in the design that the above design is what we would display if the site was using WooCommerce version **less than 4.0**.  i.e. when the new stats is not supported by the site. 

The ["try our improved stats"](https://d.pr/i/IowqFG) is only displayed if the site has WooCommerce version > 4.0 or if v4-analytics namespace is **available**. 

> Where does the "Learn More" CTA takes you on the "upgrade to keep stats" banner?

The Learn More currently takes you to the WooCommerce plugin site so that you can upgrade to the latest Woocommerce version.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt [hunk 1]
```diff
@@ -12,6 +12,7 @@ object AppUrls {
 
     const val WOOCOMMERCE_UPGRADE = "https://docs.woocommerce.com/document/how-to-update-woocommerce/"
     const val WOOCOMMERCE_ADMIN_PLUGIN = "https://wordpress.org/plugins/woocommerce-admin"
```
Comment (nbradbury) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt]: `WOOCOMMERCE_ADMIN_PLUGIN` is no longer used and can be removed.

File: WooCommerce/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -184,9 +184,10 @@
     <string name="dashboard_fulfill_order_title_multiple">You have %1$d orders to fulfill</string>
     <string name="dashboard_fulfill_orders_msg">Review, prepare, and ship these pending orders</string>
 
-    <string name="my_store_stats_reverted_message">We have reverted to the old stats. To use the beta stats, please activate the WooCommerce Admin plugin with version 0.22 or higher</string>
+    <string name="my_store_stats_reverted_title">Upgrade to keep seeing your stats</string>
+    <string name="my_store_stats_reverted_message">We\'ve rolled out improvements to our analytics. Upgrade to WooCommerce 4.0 or above or install WooCommerce Admin plugin to keep seeing your stats after Jul 1,2020</string>
```
Comment (nbradbury) [inline WooCommerce/src/main/res/values/strings.xml]: There should be a space between the date and the year, and the second sentence should end with a period.

**Comment by Garance91540** [comment]
> The Learn More currently takes you to the WooCommerce plugin site so that you can upgrade to the latest Woocommerce version.

In that case I think the CTA should read "Upgrade" not "Learn More"

**Comment by Garance91540** [comment]
Hi @anitaa1990 regarding the "try our improved stats"

> The "try our improved stats" is only displayed if the site has WooCommerce version > 4.0 or if v4-analytics namespace is available.

My understanding was that users entering in these categories will see the new stats by default, so no longer a need for the "try our improved stats". 🤔

**Comment by anitaa1990** [comment]
> My understanding was that users entering in these categories will see the new stats by default, so no longer a need for the "try our improved stats". 🤔

That would be the ideal behaviour but there are scenarios where this might not be possible. 

For instance, when the user first logs into the app, we make a call to the `wc/v4-analytics` API to determine if the store supports the new stats. The response time is generally around 0.7-1.1s by which time the old stats is already displayed to the user. Now when we find that the new stats _is_ supported, automatically refreshing the screen might be a bit jarring and this was one of the reasons the "Try our improved stats" banner was introduced. 

Similarly, everytime the app is opened, we check for the availability of the new stats. If for some reason, the user has disabled the WC-Admin stats on their store, we would no longer have access to the `v4-analytics` namespace and display the "Upgrade to keep seeing your stats" banner. 

I will DM you the link to the discussion around this when this was first implemented.

I agree that both these scenarios will not happen once we stop support for the old stats sometime around July. And when the new stats is not available, we would display this design:


But until then, we might need to display the "Try the improved stats" banner 🤔

**Review by Garance91540 — APPROVED** [review]
Looking good! Are you implementing this banner in a separate PR?

Screen Shot 2020-03-26 at 5 04 05 PM

**Comment by anitaa1990** [comment]
> Looking good! Are you implementing this banner in a separate PR?

This design informs the user if WooCommerce > 4.0 but WC-Admin plugin is disabled. But currently there is no option for us to know if the v4-analytics namespace is NOT available because user has Woo version < 4.0 or if they have the latest WooCommerce plugin but that they have disabled WC-Admin. So this PR just displays a generic error message to the user if the `v4-analytics` namespace is not available. Let me know if that makes sense!

File: WooCommerce/src/main/res/values/strings.xml (hunk 2 above)
Comment (anitaa1990) [reply WooCommerce/src/main/res/values/strings.xml]: 👍 Fixed in a19d2ae

File: WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt (hunk 1 above)
Comment (anitaa1990) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/AppUrls.kt]: 👍 Removed in a19d2ae

File: WooCommerce/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -184,9 +184,10 @@
     <string name="dashboard_fulfill_order_title_multiple">You have %1$d orders to fulfill</string>
     <string name="dashboard_fulfill_orders_msg">Review, prepare, and ship these pending orders</string>
 
-    <string name="my_store_stats_reverted_message">We have reverted to the old stats. To use the beta stats, please activate the WooCommerce Admin plugin with version 0.22 or higher</string>
+    <string name="my_store_stats_reverted_title">Upgrade to keep seeing your stats</string>
+    <string name="my_store_stats_reverted_message">We\'ve rolled out improvements to our analytics. Upgrade to WooCommerce 4.0 or above or install WooCommerce Admin plugin to keep seeing your stats after Jul 1, 2020.</string>
```
Comment (nbradbury) [inline WooCommerce/src/main/res/values/strings.xml]: I'm sorry I didn't notice this before, but this should be "July" rather than "Jul".

Comment (anitaa1990) [reply WooCommerce/src/main/res/values/strings.xml]: It was mentioned in the designs that way 🤔 But fixed in 47d03eb

**Review by nbradbury — APPROVED** [review]
:shipit:

**Pull request merged**